File: sample-app-java/src/main/java/com/weedow/spring/data/search/sample/java/entity/Task.java
package com.weedow.spring.data.search.sample.java.entity;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;

import javax.persistence.Column;
import javax.persistence.Entity;

@Entity
public class Task extends JpaPersistable<Long> {

    @Column(nullable = false)
    private String name;

    @Column
    private String description;

    public String getName() {
        return name;
    }

    public Task setName(String name) {
        this.name = name;
        return this;
    }

    public String getDescription() {
        return description;
    }

    public Task setDescription(String description) {
        this.description = description;
        return this;
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.JSON_STYLE);
    }
}

File: common/src/main/kotlin/com/weedow/spring/data/search/common/model/Task.kt
package com.weedow.spring.data.search.common.model

import org.apache.commons.lang3.builder.ToStringBuilder
import org.apache.commons.lang3.builder.ToStringStyle
import javax.persistence.Column
import javax.persistence.Entity

@Entity
class Task(
    @Column(nullable = false)
    val name: String,

    @Column
    val description: String

) : JpaPersistable<Long>() {

    // Override toString() method to get a JSON representation used by the HTTP response
    override fun toString(): String {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.JSON_STYLE)
    }
}